ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt

```diff
@@ -55,30 +64,48 @@ fun PaywallDialog(
             shouldDisplayBlock = paywallDialogOptions.shouldDisplayBlock,
         )
 
+        // This is needed because of this issue: https://issuetracker.google.com/issues/246909281.
+        // This is fixed in a newer version of Compose, but to avoid a breaking change,
+        // we are applying a workaround for now.
+        // This should be removed once we update Compose in the next major.
+        val dialogBottomPadding = if (Build.VERSION.SDK_INT > Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            WindowInsets.navigationBars.asPaddingValues().calculateBottomPadding() +
+                WindowInsets.statusBars.asPaddingValues().calculateTopPadding()
+        } else {
+            0.dp
+        }
+
         Dialog(
             onDismissRequest = {
                 dismissRequest()
                 viewModel.closePaywall()
                 paywallDialogOptions.dismissRequest?.invoke()
             },
-            properties = DialogProperties(usePlatformDefaultWidth = shouldUsePlatformDefaultWidth()),
+            properties = DialogProperties(
+                usePlatformDefaultWidth = shouldUsePlatformDefaultWidth(),
+                decorFitsSystemWindows = Build.VERSION.SDK_INT <= Build.VERSION_CODES.UPSIDE_DOWN_CAKE,
+            ),
         ) {
-            DialogScaffold(paywallOptions)
+            DialogScaffold(paywallOptions, dialogBottomPadding)
         }
     }
 }
 
 @Composable
-private fun DialogScaffold(paywallOptions: PaywallOptions) {
+private fun DialogScaffold(paywallOptions: PaywallOptions, dialogBottomPadding: Dp) {
     Scaffold(
         modifier = Modifier
             .fillMaxWidth()
             .fillMaxHeight(getDialogMaxHeightPercentage()),
+        containerColor = Color.Black.copy(alpha = 0.4f),
     ) { paddingValues ->
+        val shouldApplyDialogBottomPadding = paddingValues.calculateBottomPadding() == 0.dp &&
+            paddingValues.calculateTopPadding() == 0.dp
         Box(
             modifier = Modifier
                 .fillMaxSize()
-                .padding(paddingValues),
+                .conditional(Build.VERSION.SDK_INT <= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) { padding(paddingValues) }
+                .padding(bottom = if (shouldApplyDialogBottomPadding) dialogBottomPadding else 0.dp),
```

Comment:
is that `0.dp` maybe causing "Restore purchases" at the bottom to be almost behind the navigation bar?

Reply:
Nope, I think that's part of the paywall and was already happening in Android <35. The developer should add some margin in the editor to add a bit of spacing.